### DISCUSSION TITLE
Taro4的文档什么时候更新

### DISCUSSION THREAD

**Discussion by viilon** [NervJS/taro#16457] [Q & A]
有计划升级到taro4，但是官网没有4.0的升级指南

**Comment by honlyHuang** [comment]
4.0的文档也是有的哈 https://docs.taro.zone/docs/next/ 只是没有发正式而已，。4.0更新的内容：https://mp.weixin.qq.com/s/M7GGhgqJW1HjZHnAKf4X-A

**Reply by zhouyunjian** [reply]
。。。。。我按照文档，yarn dev:weapp编译以后（没有报错），导入项目微信小程序开发工具都没法预览。。。卡住一直没有进展，有人能给个提示么？`sudo taro init cacu
Password:
👽 Taro v4.1.3



Taro 即将创建一个新项目!
Need help? Go and open issue: https://tls.jd.com/taro-issue-helper

? 请输入项目介绍 计算器
? 请选择框架 Vue3
? 是否需要使用 TypeScript ？ Yes
? 是否需要编译为 ES5 ？ No
? 请选择 CSS 预处理器（Sass/Less/Stylus） Sass
? 请选择包管理工具 yarn
? 请选择编译工具 Vite
? 请选择模板源 Github（最新）
✔ 拉取远程模板仓库成功！
? 请选择模板 vue3-NutUI（NutUI + Vue3 模板（https://nutui.jd.com/））

✔ 创建项目: cacu
✔ 创建文件: /Users/miazhang/Documents/code/cacu/package.json
✔ 创建文件: /Users/miazhang/Documents/code/cacu/.gitignore
✔ 创建文件: /Users/miazhang/Documents/code/cacu/types/global.d.ts
✔ 创建文件: /Users/miazhang/Documents/code/cacu/types/vue.d.ts
✔ 创建文件: /Users/miazhang/Documents/code/cacu/config/index.ts
✔ 创建文件: /Users/miazhang/Documents/code/cacu/config/prod.ts
✔ 创建文件: /Users/miazhang/Documents/code/cacu/config/dev.ts
✔ 创建文件: /Users/miazhang/Documents/code/cacu/project.tt.json
✔ 创建文件: /Users/miazhang/Documents/code/cacu/project.config.json
✔ 创建文件: /Users/miazhang/Documents/code/cacu/babel.config.js
✔ 创建文件: /Users/miazhang/Documents/code/cacu/.eslintrc
✔ 创建文件: /Users/miazhang/Documents/code/cacu/tsconfig.json
✔ 创建文件: /Users/miazhang/Documents/code/cacu/.editorconfig
✔ 创建文件: /Users/miazhang/Documents/code/cacu/src/index.html
✔ 创建文件: /Users/miazhang/Documents/code/cacu/src/app.config.ts
✔ 创建文件: /Users/miazhang/Documents/code/cacu/src/app.scss
✔ 创建文件: /Users/miazhang/Documents/code/cacu/src/pages/index/index.vue
✔ 创建文件: /Users/miazhang/Documents/code/cacu/src/pages/index/index.config.ts
✔ 创建文件: /Users/miazhang/Documents/code/cacu/src/app.ts

✔ 初始化 git 成功
执行安装项目依赖 yarn install, 需要一会儿...
yarn install v1.22.22
info No lockfile found.
[1/4] Resolving packages...
warning @tarojs/helper > @swc/register@0.1.10: Use @swc-node/register instead
warning @tarojs/plugin-platform-h5 > @tarojs/service > @tarojs/runner-utils > scss-bundle > @types/sass@1.45.0: This is a stub types definition. sass provides its own type definitions, so you do not need this installed.
warning @tarojs/plugin-platform-h5 > @tarojs/service > @tarojs/runner-utils > scss-bundle > @types/glob > @types/minimatch@6.0.0: This is a stub types definition. minimatch provides its own type definitions, so you do not need this installed.
warning @tarojs/plugin-platform-h5 > @tarojs/service > @tarojs/runner-utils > scss-bundle > globs > glob@7.2.3: Glob versions prior to v9 are no longer supported
warning @tarojs/plugin-platform-h5 > @tarojs/service > @tarojs/runner-utils > scss-bundle > globs > glob > inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
warning @tarojs/cli > download-git-repo > rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
warning @tarojs/cli > download-git-repo > rimraf > glob@7.2.3: Glob versions prior to v9 are no longer supported
warning @tarojs/cli > @tarojs/plugin-doctor > eslint@8.41.0: This version is no longer supported. Please see https://eslint.org/version-support for other options.
warning @tarojs/cli > @tarojs/plugin-doctor > eslint > @humanwhocodes/config-array@0.11.14: Use @eslint/config-array instead
warning @tarojs/cli > @tarojs/plugin-doctor > eslint > file-entry-cache > flat-cache > rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
warning @tarojs/cli > @tarojs/plugin-doctor > eslint > @humanwhocodes/config-array > @humanwhocodes/object-schema@2.0.3: Use @eslint/object-schema instead
warning eslint@8.57.1: This version is no longer supported. Please see https://eslint.org/version-support for other options.
warning eslint > @humanwhocodes/config-array@0.13.0: Use @eslint/config-array instead
warning eslint > @humanwhocodes/config-array > @humanwhocodes/object-schema@2.0.3: Use @eslint/object-schema instead
[2/4] Fetching packages...

[3/4] Linking dependencies...
warning " > @tarojs/plugin-platform-weapp@4.1.3" has unmet peer dependency "@tarojs/service@4.1.3".
warning " > @tarojs/plugin-platform-alipay@4.1.3" has unmet peer dependency "@tarojs/service@4.1.3".
warning " > @tarojs/plugin-platform-tt@4.1.3" has unmet peer dependency "@tarojs/service@4.1.3".
warning " > @tarojs/plugin-platform-swan@4.1.3" has unmet peer dependency "@tarojs/service@4.1.3".
warning " > @tarojs/plugin-platform-jd@4.1.3" has unmet peer dependency "@tarojs/service@4.1.3".
warning "@tarojs/plugin-platform-h5 > @tarojs/components-react@4.1.3" has unmet peer dependency "react@*".
warning "@tarojs/plugin-platform-h5 > @tarojs/components-react@4.1.3" has unmet peer dependency "solid-js@*".
warning " > @tarojs/plugin-framework-vue3@4.1.3" has unmet peer dependency "vue-loader@^17.0.0".
warning " > @tarojs/plugin-framework-vue3@4.1.3" has unmet peer dependency "webpack@^5".
warning " > @tarojs/plugin-framework-vue3@4.1.3" has unmet peer dependency "@tarojs/runner-utils@4.1.3".
[4/4] Building fresh packages...
success Saved lockfile.
Done in 159.66s.
✔ 安装项目依赖成功
创建项目 cacu 成功！
请进入项目目录 cacu 开始工作吧！😝

yarn dev:weapp
yarn run v1.22.22
$ npm run build:weapp -- --watch

> cacu@1.0.0 build:weapp
> taro build --type weapp --watch

👽 Taro v4.1.3

Tips:
1. 预览模式生成的文件较大，设置 NODE_ENV 为 production 可以开启压缩。
Example:
$ NODE_ENV=production taro build --type weapp --watch


vite v4.5.14 building for production...

watching for file changes...

build started...
transforming (79) node_modules/@tarojs/runtime/dist/dom-external/node.jsDeprecation Warning [legacy-js-api]: The legacy JS API is deprecated and will be removed in Dart Sass 2.0.0.

More info: https://sass-lang.com/d/legacy-js-api

✓ 483 modules transformed.
dist/pages/index/index.wxml    0.08 kB
dist/pages/index/index.json    0.09 kB │ gzip:   0.11 kB
dist/comp.json                 0.11 kB │ gzip:   0.11 kB
dist/comp.wxml                 0.14 kB
dist/app.json                  0.22 kB │ gzip:   0.15 kB
dist/utils.wxs                 1.00 kB
dist/base.wxml                54.40 kB
dist/vendors.wxss            110.59 kB
dist/comp.js                   0.13 kB │ gzip:   0.12 kB
dist/app.js                    0.96 kB │ gzip:   0.62 kB
dist/pages/index/index.js      1.51 kB │ gzip:   0.74 kB
dist/babelHelpers.js           7.07 kB │ gzip:   2.11 kB
dist/vendors.js               14.68 kB │ gzip:   3.25 kB
dist/taro.js                 485.81 kB │ gzip: 115.57 kB
built in 4187ms.





`

**Comment by noneorone** [comment]
刚试了下，taro4的项目初始化后，编译运行的时候一直报错，不知道是哪里的问题，可能是vite版本的问题。

**Reply by hanjin001** [reply]
选用的哪个版本呢，报什么错呢，建议是使用 webpack5 + react + taro v4

**Reply by noneorone** [reply]
webpack5 + vue3 +taro v4.0.6，好像说是vite的版本有冲突，用vite^4版本的也有其他错误。

**Comment by xiaohp** [comment]
文档还是 react class 组件，确实不太方便

**Comment by GreatAuk** [comment]
什么原因这么难产？

**Reply by hanjin001** [reply]
[Taro 4 文档](https://docs.taro.zone/docs/)已更新。
Taro 4 和 Taro 3 文档上差别不大，主要新增了一些功能。参考以下文章。
[Taro 4.0 Beta 发布：支持开发鸿蒙应用、小程序编译模式、Vite 编译等](https://juejin.cn/post/7330792655125463067)

**Reply by wmpeng** [reply]
> [Taro 4 文档](https://docs.taro.zone/docs/)已更新。 Taro 4 和 Taro 3 文档上差别不大，主要新增了一些功能。参考以下文章。 [Taro 4.0 Beta 发布：支持开发鸿蒙应用、小程序编译模式、Vite 编译等](https://juejin.cn/post/7330792655125463067)

选择了4.x版本，文档中的内容甚至还是Taro 3

**Reply by kuma0605** [reply]
对啊，内容还是显示的Taro3，难绷

**Reply by hanjin001** [reply]
> 对啊，内容还是显示的Taro3，难绷

Taro4 更新的不算多，主要是支持 vite 和鸿蒙、基建 rust 等，Taro4 和 Taro3 文档相同比较正常的。

**Reply by singi2016cn** [reply]
![image](https://github.com/user-attachments/assets/3545b571-c986-4833-8191-692846fb934c)
看下这个，对新人来说，卡死！！！

**Comment by zhouyunjian** [comment]
建议把文档也放到github开源了，自己维护不行的，
